hazelcast/src/main/java/com/hazelcast/map/impl/operation/ContainsValueExceptKeysOperation.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation;
+
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.map.impl.record.Record;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.PartitionAwareOperation;
+import com.hazelcast.spi.ReadonlyOperation;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+public class ContainsValueExceptKeysOperation extends MapOperation implements PartitionAwareOperation, ReadonlyOperation {
+
+    private boolean contains;
+    private Object value;
+    private Set<Data> deletedKeys;
+
+    public ContainsValueExceptKeysOperation(String name, Object value, Set<Data> deletedKeys) {
+        super(name);
+        this.value = value;
+        this.deletedKeys = deletedKeys;
+    }
+
+    public ContainsValueExceptKeysOperation() {
+    }
+
+    @Override
+    public int getId() {
+        return MapDataSerializerHook.CONTAINS_VALUE_EXCEPT_KEYS;
+    }
+
+    @Override
+    public void run() throws Exception {
+        final Iterator<Record> iterator = recordStore.iterator();
```

Comment:
This is not enough, there are several additional things needed here:
- check if map is loaded
- respect the different in-memory formats when comparing values
- respect the map expiration

Please take a look at `com.hazelcast.map.impl.operation.ContainsValueOperation`, we should probably make a very similar operation. You can add a new method on the `com.hazelcast.map.impl.recordstore.RecordStore`: 
`boolean containsValue(Object testValue, Set<Data> deletedKeys);`
And then implement in a similar way as `com.hazelcast.map.impl.recordstore.DefaultRecordStore#containsValue(java.lang.Object)`. Maybe even have the existing method use the new one with a `null` parameter, e.g.
```java
    @Override
    public boolean containsValue(Object value) {
        return containsValue(value, null);
    }

    @Override
    public boolean containsValue(Object value, Set<Data> deletedKeys) {
        checkIfLoaded();
        long now = getNow();
        Collection<Record> records = storage.values();

        if (!records.isEmpty()) {
            // optimisation to skip serialisation/deserialisation
            // in each call to RecordComparator.isEqual()
            value = inMemoryFormat == InMemoryFormat.OBJECT
                    ? serializationService.toObject(value)
                    : serializationService.toData(value);
        }

        for (Record record : records) {
            if (getOrNullIfExpired(record, now, false) == null) {
                continue;
            }
            if (recordComparator.isEqual(value, record.getValue())
                    && (deletedKeys == null || !deletedKeys.contains(record.getKey()))) {
                return true;
            }
        }
        return false;
    }
```